### PR TITLE
add UTF-32 info to StreamReader ctor

### PR DESCRIPTION
**Pull request by mairaw**
For internal bug https://microsoft.visualstudio.com/OS/_workitems/edit/13259781

### CODE REVIEW

File: xml/System.IO/StreamReader.xml [hunk 1]
```diff
@@ -270,7 +270,7 @@
 ## Remarks  
  This constructor initializes the encoding to <xref:System.Text.UTF8Encoding>, the <xref:System.IO.StreamReader.BaseStream%2A> property using the `stream` parameter, and the internal buffer size to 1024 bytes.  
   
- The `detectEncodingFromByteOrderMarks` parameter detects the encoding by looking at the first three bytes of the stream. It automatically recognizes UTF-8, little-endian Unicode, and big-endian Unicode text if the file starts with the appropriate byte order marks. See the <xref:System.Text.Encoding.GetPreamble%2A?displayProperty=nameWithType> method for more information.  
+ The `detectEncodingFromByteOrderMarks` parameter detects the encoding by looking at the first three bytes of the stream. It automatically recognizes UTF-8, little-endian Unicode, little-endian UTF-32, big-endian Unicode, and big-endian UTF-32 text if the file starts with the appropriate byte order marks. See the <xref:System.Text.Encoding.GetPreamble%2A?displayProperty=nameWithType> method for more information.
```
Comment (rpetrusha) [inline xml/System.IO/StreamReader.xml]: This is a nit, but I'd change the order here: little-endian Unicode, big-endian Unicode, little-endian UTF-32, and big-endian UTF-32 text... (I find the order here much more difficult to parse.) The same comment applies to each of the changes.

**Comment by mairaw** [comment]
I now noticed that @ddv239 also mentioned on the bug that BOM is detected by looking at first four, not three, bytes of the stream. Should I also change that @tarekgh and @krwq? Thanks again!

**Comment by ddv239** [comment]
Of course everything listed in the bug should be addressed. Yes, first four bytes, not 3 (the UTF-32 BOM is 4 bytes, obviously).

**Comment by tarekgh** [comment]
@mairaw, what @ddv239 is correct. UTF32 BOM is 4-bytes. you may look at link https://en.wikipedia.org/wiki/Byte_order_mark it has a nice table for the BOM values. Note that, the framework doesn't support the BOM's in the table starting from UTF7  and below.

**Comment by mairaw** [comment]
Thanks guys! I've changed that as well. Given that msdn will be redirected soon, I'll keep just the change in docs.

**Pull request merged**